internal/orchestrator/updates.go

```diff
@@ -478,25 +492,89 @@ func (o *Orchestrator) getEventsToReconcile(ctx context.Context) (evts []*v1type
 	return evts
 }
 
+// XXX: testing note -- all the functions called here (ConditionUpdateEvent::Validate(),
+// Repository::GetActiveCondition(), and ConditionUpdateEvent::MergeExisting()) are all tested in their
+// respective modules, so I'm skipping testing this function.
+func (o *Orchestrator) mergeUpdate(ctx context.Context, updEvt *v1types.ConditionUpdateEvent) (*rctypes.Condition, error) {
+	_, span := otel.Tracer(pkgName).Start(ctx, "orchestrator.mergeUpdate")
+	defer span.End()
+
+	metrics.RegisterSpanEvent(
+		span,
+		updEvt.ServerID.String(),
+		updEvt.ConditionID.String(),
+		string(updEvt.Kind),
+		"mergeUpdate",
+	)
+
+	if err := updEvt.Validate(); err != nil {
+		o.logger.WithError(err).WithFields(logrus.Fields{
+			"server_id":      updEvt.ConditionUpdate.ServerID,
+			"condition_kind": updEvt.Kind,
+		}).Error("conditionUpdateEvent validate error")
+		return nil, err
+	}
+
+	// query existing condition
+	existing, err := o.repository.GetActiveCondition(ctx, updEvt.ConditionUpdate.ServerID)
+	if err != nil {
+		if errors.Is(err, store.ErrConditionNotFound) {
+			o.logger.WithFields(logrus.Fields{
+				"server_id":      updEvt.ConditionUpdate.ServerID,
+				"condition_id":   updEvt.ConditionUpdate.ConditionID,
+				"condition_kind": updEvt.Kind,
+			}).Error("no existing pending/active condition found for update")
+			return nil, errors.Wrap(err, "fetching active condition to update")
+		}
+		return nil, errors.Wrap(errRetryThis, err.Error())
+	}
+
```

Comment:
I think we're missing two cases,
- The update is applied - only if the existing CR State is non-final
- The status KV record for the event is purged, even the CR has no record of a matching ConditionID

This would cover the changes proposed in 
https://github.com/metal-toolbox/conditionorc/pull/230/commits/2ab1f3888243cfe46cde56a18288923297fdcac1#diff-b36bc375033af0c0f43e3fa155cb4078dc7dfba67b7b8986f95fb00a8cb2ce45R501